### PR TITLE
[docs/reference] more fixes in Markdown files

### PR DESCRIPTION
**Pull request by michelou**
Follow up of PR #10767.

Fixes further issues found in MarkDown files in [`docs/reference/`](https://github.com/lampepfl/dotty/tree/master/docs/docs/reference), e.g,
- removed extra spaces
- corrected misspellings, etc.
- reduced line width in several code blocks (to avoid text overflow in the right margin)
- added missing link to PNG image (class hierarchy in section 6.11.1)
- added several external links (e.g. Unicode character)

**PS.** The result of the above changes can be viewed in the PDF document [`scala3_reference.pdf`](https://github.com/michelou/dotty-examples/blob/master/docs/scala3_reference.pdf) (736 Kb) generated with [Pandoc 2.11](https://pandoc.org/releases.html).

### CODE REVIEW

File: docs/docs/reference/contextual/derivation-macro.md [hunk 1]
```diff
@@ -91,12 +91,13 @@ The implementation of `summonAll` as a macro can be show below assuming that we
...
-    case '[tpe *: tpes]   => derived[tpe] :: summonAll[tpes]
-    case '[EmptyTuple] => Nil
-  }
+  def summonAll[T: Type](using Quotes): List[Expr[Eq[_]]] =
+    Type.of[T] match {
+      case '[String *: tpes] => '{ summon[Eq[String]] } :: summonAll[tpes]
+      case '[Int *: tpes]    => '{ summon[Eq[Int]] }    :: summonAll[tpes]
+      case '[tpe *: tpes]    => derived[tpe] :: summonAll[tpes]
+      case '[EmptyTuple]     => Nil
+    }
```
Comment (liufengyun) [inline docs/docs/reference/contextual/derivation-macro.md:100]: @nicolasstucki : Do we still support this use case? It seems to be a valid use case.

File: docs/docs/reference/metaprogramming/macros-spec.md [hunk 2]
```diff
@@ -183,7 +183,7 @@ private def dynamicPower(n: Int, x: Double): Double =
   else if (n % 2 == 0) dynamicPower(n / 2, x * x)
   else x * dynamicPower(n - 1, x)
 ```
----
+
 This assumes a `Constant` extractor that maps tree nodes representing
 constants to their values.
```
Comment (liufengyun) [inline docs/docs/reference/metaprogramming/macros-spec.md:188]: @nicolasstucki : do we still support `Constant`?

**Review by liufengyun — APPROVED** [review]
LGTM, thanks @michelou 👍 

> added missing PNG image (class hierarchy in section 6.11.1)

Do you forget the file, I can't find it in the PR.

File: docs/docs/reference/metaprogramming/macros-spec.md (hunk 2 above)
Comment (michelou) [reply docs/docs/reference/metaprogramming/macros-spec.md:188]: @liufengyun For some reason the PNG file `explicit-nulls-type-hierarchy.png` is outside the `docs/reference/` directory (see [`images/explicit-nulls/`](https://github.com/lampepfl/dotty/tree/master/docs/images/explicit-nulls)). Before running `pandoc` I call a small preprocessor script which performs several transformations, e.g.
- replace internal relative links to anchors
- copy image files to the right place for `pandoc`

Comment (nicolasstucki) [reply docs/docs/reference/metaprogramming/macros-spec.md:188]: Yes, but that code is extremely outdated and would not compile. I would use the following version instead

----------

```scala
inline def power(x: Double, n: Int): Double =
  ${ powerExpr('x, 'n) }

private def powerExpr(n: Expr[Int], x: Expr[Double])(using Quotes): Expr[Double] =
  n.value match {
    case Some(m) => powerCode(x, m)
    case _ => '{ dynamicPower($x, $n) }
  }

private def dynamicPower(x: Double, n: Int): Double =
  if (n == 0) 1.0
  else if (n % 2 == 0) dynamicPower(x * x, n / 2)
  else x * dynamicPower(x, n - 1)
```

This assumes a `.value` that maps tree nodes representing constants to their values.

----------

Comment (michelou) [reply docs/docs/reference/metaprogramming/macros-spec.md:188]: @nicolasstucki Is the following self-contained code ok for you (NB. `powerCode` is not defined in your code) ?!

For replacement of the first code example in section "Going Further" of file [`metaprogramming/macro-spec.md`](https://github.com/lampepfl/dotty/blob/master/docs/docs/reference/metaprogramming/macros-spec.md):
```scala
import scala.quoted.Expr
import scala.quoted.Quotes

inline def power(x: Double, n: Int): Double =
  ${ powerExpr('x, 'n) }

private def powerExpr(x: Expr[Double], n: Expr[Int])(using Quotes): Expr[Double] =
  n.value match
    case Some(m) => powerExpr(x, m)
    case _ => '{ dynamicPower($x, $n) }

private def powerExpr(x: Expr[Double], n: Int)(using Quotes): Expr[Double] =
  if n == 0 then '{ 1.0 }
  else if n == 1 then x
  else if n % 2 == 0 then '{ val y = $x * $x; ${ powerExpr('y, n / 2) } }
  else '{ $x * ${ powerExpr(x, n - 1) } }

private def dynamicPower(x: Double, n: Int): Double =
  if n == 0 then 1.0
  else if n % 2 == 0 then dynamicPower(x * x, n / 2)
  else x * dynamicPower(x, n - 1)
```
For testing (not to be included): `Main.scala`
```scala
@main
def Main: Unit =
  val x = 2.0
  for n <- 1 to 10 do
    println(s"power($x, $n) = ${power(x, n)}")
```

**PS.** The paragraph directly following the above code example mentions `Constant`. I assume that text can be removed right ?!

Comment (nicolasstucki) [reply docs/docs/reference/metaprogramming/macros-spec.md:188]: You can replace it with 
```markdown
This assumes a `.value` that maps tree nodes representing constants to their values.
```

Comment (nicolasstucki) [reply docs/docs/reference/metaprogramming/macros-spec.md:188]: And use `import scala.quoted._`

Comment (michelou) [reply docs/docs/reference/metaprogramming/macros-spec.md]: @liufengyun addressed review comment.

**Comment by liufengyun** [comment]
Thanks @michelou 👍

**Comment by michelou** [comment]
@liufengyun Thanks. I will create an new PR with more changes, including Nicolas's suggestion (see above):
```
This assumes a `.value` that maps tree nodes representing constants to their values.
```

**Pull request merged**